4_compile.py

```diff
@@ -112,6 +112,10 @@ def configure_optimizers(self):
 if __name__ == "__main__":
     print(watermark(packages="torch,lightning,transformers", python=True), flush=True)
     print("Torch CUDA available?", torch.cuda.is_available(), flush=True)
+    
+    # Enable Tensor Cores
+    if torch.cuda.get_device_capability() >= (8, 0):
```

Comment:
This is probably the single most important flag to configure, we can't enable it by default since it changes the numerics of a program but at this point I am heavily considering enabling it by default

Reply:
Thanks, but for ampere GPUs, it should happen automatically I thought?

Reply:
That's unfortunately not the case by default, but you might be using them if you use cast your tensors to bf16 or tf32 or if you use AMP 

I find it helpful to run NVIDIA Nsight and look at their tensor core utilization there to make sure

Reply:
Oh this code is using AMP. I meant in an AMP context above of course. Sorry if I wasn't clear.

Reply:
Ah! Well let's see if you get any changes, otherwise might need to start profiling to see what's going on